src/Storage/Services/IDataService.cs

```diff
@@ -40,6 +40,7 @@ public interface IDataService
         /// <param name="org">The application owner id.</param>
         /// <param name="stream">Data to be written to blob storage.</param>
         /// <param name="dataElement">The data element to insert.</param>
-        Task UploadDataAndCreateDataElement(string org, Stream stream, DataElement dataElement);
+        /// <param name="internalId">The internal id of the data element to insert.</param>
```

Comment:
do we need a better description of this concept of internal id? db id might be more descriptive? 
Having consequences of database limitations exposed in the service layer isn't optimal, but I guess we don't have any other options here?

Reply:
I see your concern, but I think the major problem here are all the different names of the existing keys – id, instanceid, instanceguid etc. To me this is very confusing, and I think it’s more important to change these names after cosmos is removed. In Altinn II the naming convention is id for primary key and <table>id for foreign key. I think this is a good convention, but it’s impossible to use now without changing the existing key names.